flutter/lib/src/widgets_binding_observer.dart

```diff
@@ -0,0 +1,195 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'sentry_flutter_options.dart';
+import '../sentry_flutter.dart';
+
+/// This is a `WidgetsBindingObserver` which can observe some events of a
+/// Flutter application.
+/// These are for example events related to its lifecycle, accessibility
+/// features, display features and more.
+///
+/// The tracking of each event can be configured via [SentryFlutterOptions]
+///
+/// This class calls `WidgetsBinding.instance` but we don't need to call
+/// `WidgetsFlutterBinding.ensureInitialized()` because we can only add this
+/// class to an instance of `WidgetsBinding`.
+///
+/// See also:
+///   - [WidgetsBindingObserver](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver-class.html)
+class SentryWidgetsBindingObserver with WidgetsBindingObserver {
+  SentryWidgetsBindingObserver({
+    Hub hub,
+    @required SentryFlutterOptions options,
+  }) {
+    this.hub = hub ?? HubAdapter();
+    assert(options != null);
+    this.options = options;
+  }
+
+  Hub hub;
+  SentryFlutterOptions options;
+
+  /// This method records lifecycle events.
+  /// It tries to mimic the behavior of ActivityBreadcrumbsIntegration of Sentry
+  /// Android for lifecycle events.
+  ///
+  /// On Android and iOS this records lifecycle event breadcrumbs which loosely
+  /// correspond to their respectiv platforms lifecycle events.
+  ///
+  /// See also:
+  ///   - [WidgetsBindingObserver](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver-class.html)
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
```

Comment:
I'm trying to reproduce this breadcrumb but this method is never been called.

And yes I do call: `options.useFlutterBreadcrumbTracking()`

ideas?

Reply:
With the main method of the included example app looking like this:
```dart
Future<void> main() async {
  await SentryFlutter.init(
    (options) {
      options.dsn = _exampleDsn;
      // use breadcrumb tracking of WidgetsBindingObserver
      options.useFlutterBreadcrumbTracking();
      // use breadcrumb tracking of platform Sentry SDKs
      // calling the following disabled the Flutter tracking
      // options.useNativeBreadcrumbTracking();
    },
    // Init your App.
    appRunner: () => runApp(MyApp()),
  );
}
```

it gets called in at least the following cases:
- App is open, in foreground and the notification bar gets dragged down (iOS)
- App is brought to the background by pressing the home button (or the swipe action on devices without a button) (Android, iOS)
- App is in foreground and the device gets locked (Android)

If it runs on Flutter for Web I'm note sure if it gets called at all.

Maybe you need to do clean build or a hot restart instead of a hot reload? 
A simple hot reload would not add the integration and observer if the app is already running.

Reply:
<img width="1013" alt="Screenshot 2020-12-04 at 11 32 43" src="https://user-images.githubusercontent.com/5731772/101153224-6e2b8300-3624-11eb-8e45-1ae15498400b.png">

probably a caching issue, yes, thanks!